reactor-netty-http-brave/src/main/java/reactor/netty/http/client/TracingContext.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.client;
+
+import brave.Span;
+import reactor.util.context.ContextView;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static reactor.netty.http.client.BraveHttpClientTracing.KEY_TRACING_CONTEXT;
+
+/**
+ * {@link TracingContext} provides the current span utilizing the reactive {@link ContextView}.
+ *
+ * {@code
+ *     Mono.deferContextual(Mono::just)
+ *         .flatMap(ctx -> TracingContext.of(ctx)
+ *                                       .ifPresent(tracing -> tracing.span(span -> ...)));
+ * }
+ *
+ * @author Violeta Georgieva
+ * @since 1.0.0
+ */
+public interface TracingContext {
```

Comment:
public interface, so I guess it could be shared with the server, given that it has no hard dependency on server/client concepts?

Reply:
Actually the key in the reactor context is different. So `TracingContext.of` gives you different things?

Reply:
right. but do we really need to support a `ContextView` containing both client and server variants of `TracingContext`? independently, maybe the logic can be mutualized and the actual key be made into a protected or private field of the `TracingContext` object, which would also encapsulate the logic of putting itself in a `Context`? (logic at https://github.com/reactor/reactor-netty/pull/1317/files#diff-e671f68ead214343939c014dc7033618R172-R175)

Reply:
ah sorry just saw that logic is around `TraceContext` from Brave, not `TracingContext` :(

Reply:
you may have one span for the server but then if you make several remote calls you will have several spans for the client.